lang/rust/Portfile

```diff
@@ -387,19 +393,20 @@ subport rust-src {
 
     use_configure no
 
+    post-extract {
+        # delete the test directories (which for some god awful reason contains binaries)
+        system -W ${worksrcpath} "find src/llvm-project -type d -name test -print0 | xargs -0 rm -rf"
+    }
+
     build {}
 
     destroot {
         xinstall -d ${rust_source_dir}
-        move ${worksrcpath}/src ${rust_source_dir}/src
-        move ${worksrcpath}/library ${rust_source_dir}/library
+        move {*}[glob ${worksrcpath}/*] ${rust_source_dir}
 
         # correct the permissions
-        system -W ${rust_source_dir} "find . -type d -exec chmod 755 {} \\;"
-        system -W ${rust_source_dir} "find . -type f -exec chmod 644 {} \\;"
-
-        # delete the test directories (which for some god awful reason contains binaries)
-        system -W ${rust_source_dir} "find . -type d -name test -print0 | xargs -0 rm -rf"
```

Comment:
Do we no longer need to deal with test directories?

Reply:
@herbygillot I've moved it to `post-extract` and limited to `test` for `src/llvm-project` which contains the binaries which are created an issue with `Scanning binaries for linking errors` phase:
```
--->  Scanning binaries for linking errors
Warning: Error parsing file /opt/local/lib/rustlib/src/rust/src/llvm-project/llvm/test/Object/Inputs/macho-invalid-dylib-small: Premature end of data, possibly corrupt file
Warning: Error parsing file /opt/local/lib/rustlib/src/rust/src/llvm-project/llvm/test/Object/Inputs/macho-invalid-header: Premature end of data, possibly corrupt file
Warning: Error parsing file /opt/local/lib/rustlib/src/rust/src/llvm-project/llvm/test/Object/Inputs/macho-invalid-rpath-small: Premature end of data, possibly corrupt file
Warning: Error parsing file /opt/local/lib/rustlib/src/rust/src/llvm-project/llvm/test/Object/Inputs/macho-universal-bad2.x86_64.i386: Premature end of data, possibly corrupt file
Warning: Error parsing file /opt/local/lib/rustlib/src/rust/src/llvm-project/llvm/test/Object/Inputs/macho64-invalid-incomplete-load-command: Premature end of data, possibly corrupt file
Warning: Error parsing file /opt/local/lib/rustlib/src/rust/src/llvm-project/llvm/test/Object/Inputs/macho64-invalid-incomplete-load-command.1: Premature end of data, possibly corrupt file
Warning: Error parsing file /opt/local/lib/rustlib/src/rust/src/llvm-project/llvm/test/Object/Inputs/macho64-invalid-incomplete-segment-load-command: Premature end of data, possibly corrupt file
--->  Found 40 broken files, matching files to ports     
--->  Found 1 broken port, determining rebuild order
```

Anyway, current `rust-src` can be installed and used ;)